src/org/freedesktop/gstreamer/Pad.java

```diff
@@ -518,6 +529,36 @@ protected void disconnect() {
     public void removeEventProbe(EVENT_PROBE listener) {
         removeCallback(EVENT_PROBE.class, listener);
     }
+    
+    
+    public synchronized void addDataProbe(final DATA_PROBE listener) {
+        
+        final GstPadAPI.PadProbeCallback probe = new GstPadAPI.PadProbeCallback() {
+            public PadProbeReturn callback(Pad pad, GstPadProbeInfo probeInfo, Pointer user_data) {
+                if ((probeInfo.padProbeType & GstPadAPI.GST_PAD_PROBE_TYPE_BUFFER) != 0) {
+                    Buffer buffer = GSTPAD_API.gst_pad_probe_info_get_buffer(probeInfo);
+                    return listener.dataReceived(pad, buffer);
+                }
+
+            	//We have to negate the return value to keep consistency with gstreamer's API
```

Comment:
What exactly does this comment mean?

Reply:
that is actually a copied comment from addEventProbe. no idea what it means :D